src/main/java/com/conveyal/gtfs/loader/ConditionalRequirement.java

```diff
@@ -0,0 +1,328 @@
+    public ConditionalCheckType referenceFieldCheck;
+    /** The minimum reference field value if a range check is being performed. */
+    public int minReferenceValue;
+    /** The maximum reference field value if a range check is being performed. */
+    public int maxReferenceValue;
+    /** The type of check to be performed on the dependent field. */
+    public ConditionalCheckType dependentFieldCheck;
+    /** The name of the dependent field, which is a field that requires a specific value if the reference and
+     * (in some cases) dependent field checks meet certain conditions.*/
+    public String dependentFieldName;
+    /** The expected dependent field value. */
+    public String dependentFieldValue;
+    /** The reference table name required for checking foreign references. */
+    String referenceTableName;
+
+    public ConditionalRequirement(
+        int minReferenceValue,
+        int maxReferenceValue,
+        String dependentFieldName,
+        ConditionalCheckType dependentFieldCheck,
+        ConditionalCheckType referenceFieldCheck,
+        String dependentFieldValue,
+        String referenceTableName
+    ) {
+        this.minReferenceValue = minReferenceValue;
+        this.maxReferenceValue = maxReferenceValue;
+        this.dependentFieldName = dependentFieldName;
+        this.dependentFieldCheck = dependentFieldCheck;
+        this.referenceFieldCheck = referenceFieldCheck;
+        this.dependentFieldValue = dependentFieldValue;
+        this.referenceTableName = referenceTableName;
+    }
+
+    public ConditionalRequirement(
+        String dependentFieldName,
+        ConditionalCheckType referenceFieldCheck
+    ) {
+        this(0, 0, dependentFieldName, null, referenceFieldCheck, null, null);
+    }
+
+    public ConditionalRequirement(
+        String dependentFieldName,
+        ConditionalCheckType referenceFieldCheck,
+        String referenceTableName
+    ) {
+        this(0, 0, dependentFieldName, null, referenceFieldCheck, null, referenceTableName);
+    }
+
+    public ConditionalRequirement(
+        int minReferenceValue,
+        int maxReferenceValue,
+        String dependentFieldName,
+        ConditionalCheckType dependentFieldCheck,
+        ConditionalCheckType referenceFieldCheck
+
+    ) {
+        this(minReferenceValue, maxReferenceValue, dependentFieldName, dependentFieldCheck, referenceFieldCheck, null, null);
+    }
+
+    public ConditionalRequirement(
+        String dependentFieldName,
+        ConditionalCheckType dependentFieldCheck,
+        ConditionalCheckType referenceFieldCheck
+    ) {
+        this(0, 0, dependentFieldName, dependentFieldCheck, referenceFieldCheck, null, null);
+    }
+
+    public ConditionalRequirement(
+        String dependentFieldName,
+        String dependentFieldValue,
+        ConditionalCheckType referenceFieldCheck
+    ) {
+        this(0, 0, dependentFieldName, null, referenceFieldCheck, dependentFieldValue, null);
+    }
```

Comment:
With so many different constructors here, it seems like it might make sense to make the `ConditionalRequirement` an abstract class with each of these constructors potentially being an implementing class. This could also be taken a step further to have a `check(LineContext lineContext, Field referenceField, TreeMultimap<String, String> uniqueValuesForFields)` abstract method that each implementing class could implement. In doing this, that big switch statement in the ReferenceTracker could be removed entirely in favor of calling `conditionalRequirement.check(...)`.

Reply:
All of the following static methods here would be the respective `check` method in the implementing classes.

Reply:
Updated as suggested. Not 100% on the new class names! Happy to update if requried.